accern_xyme/accern_xyme.py

```diff
@@ -751,6 +753,46 @@ def get_settings(self) -> Dict[str, Any]:
         assert self._settings is not None
         return self._settings
 
+    def get_timing(
+                self,
+                blacklist: Optional[List[str]] = None,
+                ) -> Optional[TimingResult]:
+        blacklist = [] if blacklist is None else blacklist
+        node_timing: Dict[str, NodeTiming] = {}
+        nodes = self.get_nodes()
+
+        def filter_blacklist(node_time: List[Timing]) -> Iterator[Timing]:
+            for value in node_time:
+                if value["name"] != blacklist:
+                    yield value
+
+        for node in nodes:
+            node_get = self.get_node(node)
+            node_time = node_get.get_timing()
+            node_name: str = node_get.get_node_def()["name"]
+            node_id = node_get.get_id()
+            node_total = 0.0
+            filter_node = None
+            for filter_node in filter_blacklist(node_time):
+                length: int = len(list(filter_blacklist(node_time)))
+                node_total += float(filter_node["total"])
+            node_timing[node_id] = {
+                "node_name": node_name,
+                "node_total": node_total,
+                "node_avg": node_total / length,
+                "fns": [filter_node],
+            }
+            pipe_total = 0.0
```

Comment:
pipe_total must be initialized before the main loop that goes over all the nodes -- how else can you compute it?